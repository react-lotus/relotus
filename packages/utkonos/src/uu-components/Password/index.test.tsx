/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { Password } from '.';

describe('UIKIT  Password input', () => {
  describe('snapshots', () => {
    it('default snapshot', () => {
      const tree = renderer.create(<Password />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('with lable snapshot', () => {
      const tree = renderer.create(<Password label="test" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('whith error', () => {
      const tree = renderer.create(<Password name="testName" error="Test error" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
