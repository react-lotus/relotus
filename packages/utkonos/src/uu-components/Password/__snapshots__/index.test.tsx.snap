// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIKIT  Password input snapshots default snapshot 1`] = `
<div
  className="uu-password"
>
  <input
    className="uu-passwordInput"
    id="nanoid"
    type="password"
  />
  <button
    className="uu-passwordShowIcon"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="uu-passwordShowIconSvg"
    >
      eyeCrossed.svg
    </svg>
  </button>
</div>
`;

exports[`UIKIT  Password input snapshots whith error 1`] = `
<div
  className="uu-password"
>
  <input
    className="uu-passwordInput uu-passwordInput_hasError"
    id="nanoid"
    name="testName"
    type="password"
  />
  <div
    className="uu-passwordError uu-inputError"
  >
    <svg
      className="uu-inputErrorIcon uu-tooltipRoot uu-popoverRoot"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      error.svg
    </svg>
  </div>
  <button
    className="uu-passwordShowIcon"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="uu-passwordShowIconSvg"
    >
      eyeCrossed.svg
    </svg>
  </button>
</div>
`;

exports[`UIKIT  Password input snapshots with lable snapshot 1`] = `
<div
  className="uu-labeled_direction_column"
>
  <label
    aria-label="test"
    className="uu-labeledLabel_direction_column uu-label uu-label_text"
    htmlFor="nanoid"
  >
    test
  </label>
  <div
    className="uu-password"
  >
    <input
      className="uu-passwordInput"
      id="nanoid"
      type="password"
    />
    <button
      className="uu-passwordShowIcon"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="uu-passwordShowIconSvg"
      >
        eyeCrossed.svg
      </svg>
    </button>
  </div>
</div>
`;
